core/vm/contracts.go

```diff
@@ -38,8 +38,8 @@ import (
 // requires a deterministic gas count based on the input size of the Run method of the
 // contract.
 type PrecompiledContract interface {
-	RequiredGas(input []byte) uint64  // RequiredPrice calculates the contract gas use
-	Run(input []byte) ([]byte, error) // Run runs the precompiled contract
+	RequiredGas(input []byte) uint64                   // RequiredPrice calculates the contract gas use
+	Run(statedb StateDB, input []byte) ([]byte, error) // Run runs the precompiled contract
```

Comment:
The `StateDB` is a huge interface. To consider: do we want to use that full interface here? The only thing we need, so far, is 
```
	GetState(common.Address, common.Hash) common.Hash
```
So we could make a smaller interface, which would be easier to mock up for testing.